drivers: gpio: pca-series: Reset device before register access

ct-lt opened the pull request:
Reset the port expander before trying to access any register on initialization.

This ensures the device is in a known state before accessing registers and that the initialized cache accurately reflects the post-reset state of the device.
Additionally, the reset forces the port expander to determine its address right before the first I2C transaction. This is important in setups where the address selection depends on something that may change after power-up like GPIOs (This is the case for our custom board, where we have SDA and SCL pull-ups tied to GPIOs).

While at it, I also casted the pointers to be printed with `%p` to `void*` to silence the warnings.

henrikbrixandersen reviewed (changes requested):
> While at it, I also casted the pointers to be printed with `%p` to `void*` to silence the warnings.

Looks good, but please separate this last bit into its own commit.

ct-lt committed:
drivers: gpio: pca-series: Reset device before register access

Reset the port expander before accessing any register
or updating the cache.

Signed-off-by: Lars Thiemann <thiemann@cognid.de>

ct-lt committed:
drivers: gpio: pca-series: Cast pointer to void * in logging statements

This silences warnings created at runtime, as the %p format specifier
expects a void pointer.

Signed-off-by: Lars Thiemann <thiemann@cognid.de>
